src/concretes/peer.ts

```diff
@@ -131,16 +152,55 @@ export class Peer implements IPeer {
             }
             for (const peer of peers) {
                 // Check if peer is online and try to join
-                const message = new Message(MessageType.JOIN, this.connectionService.GUID, JSON.stringify({ip: THIS_IP}));
-                await this.connectionService.sendMessageByIp(message, peer,
-                    async (responseMessage: Message) => { await this.joinResponse(responseMessage); });
+                const message = new Message(MessageType.JOIN, this.connectionService.GUID, JSON.stringify({type: this.type}));
```

Comment:
Change the name of 'type', it is not clear if this is referencing to the messageType or peerType.